addons/web_editor/static/lib/odoo-editor/src/commands/deleteBackward.js

```diff
@@ -164,7 +164,16 @@ HTMLElement.prototype.oDeleteBackward = function (offset, alreadyMoved = false)
 
     let node = this.childNodes[offset];
     let firstBlockIndex = offset;
-    while (node && !isBlock(node)) {
+
+    // `moveNodeLimit` will ensure we never move nodes that were not initialy in the element
+    //  => when Deleting and merging an element the containing node will temporary be hosted
+    //  in the common parent beside possible other nodes. We don't want to touch those others node when merging
+    //  two html elements
+    //  ex : <div>12<p>ab[]</p><p>cd</p>34</div> should never touche the 12 and 34 text node.
+    while (
+        node && !isBlock(node) &&
+        (moveNodeLimit === null || offset + moveNodeLimit > firstBlockIndex)
+    ) {
```

Comment:
Wouldn't it be possible to directly give to the parent the proper offset to use for `firstBlockIndex` in this case ? We could have:
```js
HTMLElement.prototype.oDeleteBackward = function (offset, alreadyMoved = false, firstBlockIndex = offset) {
```
and calling it like this below:
```js
cursorNode.oDeleteBackward(cursorOffset, alreadyMoved, cursorOffset + firstBlockIndex - offset);
```
... ah but maybe that's going to be wrong if `node` is a block though... it just seems very weird to me to pass an `offset` to the function then pass another parameter to _correct_ that offset later on in the function body... it would make more sense to me to send in the correct offset in the first place, though I might be missing something... what do you think ?

Reply:
I've change it to be closer to your suggestion.
What do you reckon ?